### PR TITLE
Resolve Chisel Deprecations 

### PR DESCRIPTION
**Pull request by davidbiancolin**
Subsumes #779. I have to pull this into local for CI to run. h/t @sequencer

### CODE REVIEW

**Comment by sequencer** [comment]
Seems my bad...

**Comment by sequencer** [comment]
Emmm, I tried to play with the build system, failed to understand that again.

**Comment by davidbiancolin** [comment]
Seems like an easy fix though.

**Comment by davidbiancolin** [comment]
Ok i can clean this up later today, haha.

**Comment by sequencer** [comment]
lol thanks!

File: sim/midas/src/main/scala/midas/FPGAQoRShimGenerator.scala [hunk 1]
```diff
@@ -35,41 +35,3 @@ class Midas2QoRTargets extends Config((site, here, up) => {
...
-  val exOptions = new ExecutionOptionsManager("qor")
-    with HasChiselExecutionOptions
-    with HasFirrtlOptions
-    with HasUnitTestOptions
-
-  exOptions.parse(args)
-
-  val params = getConfig(exOptions.qorOptions.fullConfigClasses).toInstance
-  Driver.execute(exOptions, () => new QoRShim()(params))
-}
```
Comment (timsnyder-siv) [inline sim/midas/src/main/scala/midas/FPGAQoRShimGenerator.scala:75]: What's the deal with this being removed?  Is it dead code that we just didn't want to update for stage/phase?

**Review by timsnyder-siv — APPROVED** [review]
I understand everything except the removal of QoRShimGenerator but I trust that @davidbiancolin knows why that can be deleted and approve.

**Comment by timsnyder-siv** [comment]
@ingallsj in case you get into a mood to clean up deprecations and other warnings, this big cleanup in FireSim is on it's way.

@sequencer thank you so much for contributing this!

File: sim/midas/src/main/scala/midas/FPGAQoRShimGenerator.scala (hunk 1 above)
Comment (sequencer) [reply sim/midas/src/main/scala/midas/FPGAQoRShimGenerator.scala:75]: `ExecutionOptionsManager` is deleted from FIRRTL.
Basically, I should re-implement this trait, but I'm not so sure, the usability of this trait, so I directly deleted.

**Comment by sequencer** [comment]
Dont merge yet, CI is still failing on my side:
https://github.com/sequencer/chipsalliance-playground/actions/runs/944337170

**Comment by davidbiancolin** [comment]
> https://github.com/sequencer/chipsalliance-playground/actions/runs/944337170

If i had to guess it's because i reintroduced a deprecated call to Driver, since there wasn't a migration path yet. 
i.e., we need this: chipsalliance/chisel3#1854

**Comment by sequencer** [comment]
> https://github.com/sequencer/chipsalliance-playground/actions/runs/946958295

passed in Chisel Firrtl x.5 version.

**Pull request merged**